src/Microsoft.Extensions.DependencyModel/Resolution/PackageCacheCompilationAssemblyResolver.cs

```diff
@@ -42,37 +41,19 @@ public bool TryResolveAssemblyPaths(CompilationLibrary library, List<string> ass
 
             if (!string.IsNullOrEmpty(_packageCacheDirectory))
             {
-                var hashSplitterPos = library.Hash.IndexOf('-');
-                if (hashSplitterPos <= 0 || hashSplitterPos == library.Hash.Length - 1)
-                {
-                    throw new InvalidOperationException($"Invalid hash entry '{library.Hash}' for package '{library.Name}'");
-                }
-
                 string packagePath;
                 if (ResolverUtils.TryResolvePackagePath(_fileSystem, library, _packageCacheDirectory, out packagePath))
                 {
-                    string cacheHashFileName = library.HashPath;
-                    if (string.IsNullOrEmpty(cacheHashFileName))
-                    {
-                        var hashAlgorithm = library.Hash.Substring(0, hashSplitterPos);
-                        cacheHashFileName = $"{library.Name}.{library.Version}.nupkg.{hashAlgorithm}";
-                    }
-                    var cacheHashPath = Path.Combine(packagePath, cacheHashFileName);
-
-                    if (_fileSystem.File.Exists(cacheHashPath) &&
-                        _fileSystem.File.ReadAllText(cacheHashPath) == library.Hash.Substring(hashSplitterPos + 1))
-                    {
-                        assemblies.AddRange(ResolverUtils.ResolveFromPackagePath(_fileSystem, library, packagePath));
-                        return true;
-                    }
+                    assemblies.AddRange(ResolverUtils.ResolveFromPackagePath(_fileSystem, library, packagePath));
+                    return true;
                 }
             }
             return false;
         }
 
         internal static string GetDefaultPackageCacheDirectory(IEnvironment environment)
         {
-            return environment.GetEnvironmentVariable("DOTNET_PACKAGES_CACHE");
+            return environment.GetEnvironmentVariable("DOTNET_HOSTING_OPTIMIZATION_CACHE");
```

Comment:
I see that now.  

My plan is that I delete this whole file in this change. Then when @ramarag makes his change (referenced above) he will make this whole scenario work correctly with the new cache feature.